sound/soc/sof/sof-audio.c

```diff
@@ -181,12 +189,34 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (++swidget->use_count > 1)
 		return 0;
 
+	/*
+	 * The scheduler widget for a pipeline is not part of the connected DAPM
+	 * widget list and it needs to be set up before the widgets in the pipeline
+	 * are set up. The use_count for the scheduler widget is incremented for every
```

Comment:
Hrm, according to the comment, the scheduler widget needs to be set up before any other widgets for the dynamic pipeline, so in case of a scheduler widget we skip the `sof_widget_setup(sdev, swidget->pipe_widget);`

What is the connection between we need to have the scheduler widget as first to set up and skipping the swidget->pipe_widget setup?